mmcv/utils/config.py

```diff
@@ -120,6 +122,51 @@ def _substitute_predefined_vars(filename, temp_config_name):
         with open(temp_config_name, 'w') as tmp_config_file:
             tmp_config_file.write(config_file)
 
+    @staticmethod
+    def _pre_substitute_base_vars(filename, temp_config_name):
+        """Substitute base variable placehoders to string, so that parsing
+        would work."""
+        with open(filename, 'r', encoding='utf-8') as f:
+            # Setting encoding explicitly to resolve coding issue on windows
+            config_file = f.read()
+        base_var_dict = {}
+        regexp = r'\{\{\s*base\.([\w\.]+)\s*\}\}'
+        base_vars = set(re.findall(regexp, config_file))
+        for base_var in base_vars:
+            randstr = f'_{base_var}_{uuid.uuid4().hex.lower()[:6]}'
+            base_var_dict[randstr] = base_var
```

Comment:
Why we need randstr, for json/yaml config? We may add comments to explain these.

Reply:
They are temporary names used for varaible replacement. Adding some noise to reduce the probability of replacing strings that happens to be the same string

Reply:
Will add a test case for this